Modix.Services/Core/MessageLogBehavior.cs

```diff
@@ -54,6 +62,44 @@ private async Task<bool> ShouldSkip(IGuild guild, IMessageChannel channel)
             return result;
         }
 
+        private async Task HandleMessageReceived(SocketMessage message)
+        {
+            Log.LogDebug("Handling message received event for message #{MessageId}.", message.Id);
+
+            if (!message.Content.StartsWith('!') &&
+                message.Channel is IGuildChannel channel &&
+                message.Author is IGuildUser author &&
+                !author.IsBot && !author.IsWebhook)
+            {
+                await SelfExecuteRequest<ModixContext>(async db =>
```

Comment:
There should be no interaction with `ModixContext` outside of `Modix.Data`. You need to write a repository interface and class.

Reply:
I started to, but it introduced a lot of additional ceremony without a clear benefit. I opted for the simplest approach.

Reply:
It's indeed a lot of tedium, but I feel it needs to be there because:

A) It's the clearly established pattern.
B) Dependencies against a `DbContext` object are EXTREMELY difficult to decouple. When we start adding testing, anything with a `DbContext` dependency is going to be almost impossible to write tests for. Any tests we DO write are probably going to involve building a fake database of some kind, so they'll still be convoluted.
C) The difference between a good EF query and a terrible EF query are very subtle (regarding how they're translated to SQL), so it's extremely beneficial to keep tight control over where and how EF queries are being written. Having individual queries be well defined as a single method will also make it much easier to test, and even automatically find queries that aren't being translated properly (I.E. we can actually check how many SQL queries were executed).
D) All of the above was discussed at length before any of it was committed into Modix.